fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/controller/FinTechAuthorizationImpl.java

```diff
@@ -57,19 +57,24 @@
     }
 
     @Override
-    public ResponseEntity fromConsentOkGET(String authId, String finTechRedirectCode, UUID xRequestID, String xsrfToken) {
-        if (consentService.confirmConsent(authId, xRequestID)) {
-            authorizeService.getSession().setConsentConfirmed(true);
+    public ResponseEntity<Void> fromConsentOkGET(String authId, String okOrNotok, String finTechRedirectCode, UUID xRequestID, String xsrfToken) {
+        if (!authorizeService.isAuthorized()) {
+            log.warn("fromConsentOkGET failed: user is not authorized! - path is \"/v1/" + authId + "/fromConsentOk\"");
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+
+        if ("ok".equalsIgnoreCase(okOrNotok) && consentService.confirmConsent(authId, xRequestID)) {
```

Comment:
as I understood `okOrNotok` is a state of some session. And in this case, the boolean type will be enough in order to recognize 'ok' (true) or `notOk` (false) state. 
But maybe using an enum as an indicator of state will be preferred. Enum value could be compared with the state without hardcoded string value.

Reply:
Ok, changed it to enum in api, impl and ui. But api enum is not generated, to I had to create enum value by hand (https://github.com/OpenAPITools/openapi-generator/issues/2834)